library/sha256.c

```diff
@@ -37,6 +37,33 @@
 #if defined(__aarch64__)
 #  if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT) || \
     defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+/* *INDENT-OFF* */
+#    if !defined(__ARM_FEATURE_CRYPTO)
+#      if defined(__clang__)
+#        if __clang_major__ < 18
+           /* TODO: Re-consider above after https://reviews.llvm.org/D131064
+            *       merged.
+            *
+            * The intrinsic declaration are guarded with ACLE predefined macros
+            * in clang, and those macros are only enabled with command line.
+            * Define the macros can enable those declaration and avoid compile
+            * error on it.
+            */
+#          define __ARM_FEATURE_CRYPTO 1
+#        endif
+#        pragma clang attribute push (__attribute__((target("crypto"))), apply_to=function)
+#        define MBEDTLS_POP_TARGET_PRAGMA
```

Comment:
[change-request] This macro is not used anywhere. I guess the intent was to allow a pop? But it's incomplete, and since we can't do that for GCC anyway, there's no point in doing it for clang.

Reply:
Sorry, I forget that after re-writing the history

Reply:
Add it back. 

Also I add `#pragma GCC push_options`. I think  behavior of target MUST be same for both.